contracts/truefi2/TrueFiPool2.sol

```diff
@@ -0,0 +1,408 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.10;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import {ERC20} from "../common/UpgradeableERC20.sol";
+import {Ownable} from "../common/UpgradeableOwnable.sol";
+
+import {ITrueStrategy} from "./interface/ITrueStrategy.sol";
+import {ITrueFiPool2} from "./interface/ITrueFiPool2.sol";
+import {ITrueLender} from "./interface/ITrueLender.sol";
+import {ABDKMath64x64} from "../truefi/Log.sol";
+
+/**
+ * @title TrueFiPool2
+ * @dev Lending pool which may use a strategy to store idle funds
+ * Earn high interest rates on currency deposits through uncollateralized loans
+ *
+ * Funds deposited in this pool are not fully liquid. Liquidity
+ * Exiting the pool has 2 options:
+ * - withdraw a basket of LoanTokens backing the pool
+ * - take an exit penalty depending on pool liquidity
+ * After exiting, an account will need to wait for LoanTokens to expire and burn them
+ * It is recommended to perform a zap or swap tokens on Uniswap for increased liquidity
+ *
+ * Funds are managed through an external function to save gas on deposits
+ */
+contract TrueFiPool2 is ITrueFiPool2, ERC20, Ownable {
+    using SafeMath for uint256;
+
+    // ================ WARNING ==================
+    // ===== THIS CONTRACT IS INITIALIZABLE ======
+    // === STORAGE VARIABLES ARE DECLARED BELOW ==
+    // REMOVAL OR REORDER OF VARIABLES WILL RESULT
+    // ========= IN STORAGE CORRUPTION ===========
+
+    uint8 public constant VERSION = 0;
+
+    IERC20 public currencyToken;
```

Comment:
Rename it to token so that id doesn't suggest it has something to do with TrueCurrency